tools/commands-packages-query.js

```diff
@@ -0,0 +1,1159 @@
+// These commands deal with aggregating local package data with the information
+// contained in the Meteor Package Server. They also deal with presenting this
+// to the user in various human or machine-readable ways.
+var _ = require('underscore');
+var archinfo = require('./archinfo.js');
+var buildmessage = require('./buildmessage.js');
+var catalog = require('./catalog.js');
+var Console = require("./console.js").Console;
+var files = require('./files.js');
+var isopackets = require('./isopackets.js');
+var main = require('./main.js');
+var projectContextModule = require('./project-context.js');
+var utils = require('./utils.js');
+
+// We want these queries to be relatively fast, so we will only refresh the
+// catalog if it is > 15 minutes old
+var DEFAULT_MAX_AGE_MS = 15 * 60 * 1000;
+
+// Maximum number of recent versions of a package or a release that we should
+// return to the user, unless a more complete mode is requested.
+var MAX_RECENT_VERSIONS = 5;
+
+// XXX: Remove this if/when we do a Troposphere migration to backfill release
+// version publication times.
+// Estimate the publication date for a release. Since we have failed to keep
+// track of publication times of release versions in the past, we will try to
+// guess that the release was published at the same time as the tool.
+var getReleaseVersionPublishedOn = function (versionRecord) {
+  if (versionRecord.published) {
+    return new Date(versionRecord.published);
+  }
+  // We don't know when the release was published. Luckily, since there is no
+  // way to use the tool outside of a release, and we always change the tool
+  // between releases, it is a good bet that the release was published on the
+  // same day as the tool.
+  var toolPackage = versionRecord.tool.split('@');
+  var toolName = toolPackage[0];
+  var toolVersion = toolPackage[1];
+  var toolRecord = catalog.official.getVersion(toolName, toolVersion);
+  return new Date(toolRecord.published);
+};
+
+// Convert package version dependencies into a more helpful format for output in
+// 'meteor show'.
+var formatDependencies = function (versionDependencies) {
+  var dependencies = _.map(
+    // The dependency on 'meteor' was almost certainly added automatically, by
+    // Isobuild. Returning this to the user will only cause confusion.
+    _.omit(versionDependencies, "meteor"),
+    function (dep, depName) {
+      // We will only consider this a weak dependency if all of its references
+      // are marked as weak.
+      var weak = _.every(dep.references, function (ref) {
+        return !! ref.weak;
+      });
+      return {
+        name: depName,
+        constraint: dep.constraint,
+        weak: weak
+      };
+  });
+  return dependencies;
+};
+
+// Converts an object to an EJSON string with the right spacing.
+var convertToEJSON = function (data) {
+  var EJSON = isopackets.load('ejson').ejson.EJSON;
```

Comment:
We have been able to have EJSON isopackets since we separated the former-uniload-catalog out of the remote-catalog BITD! (We couldn't do it before, because we needed to read the remote catalog data in order to load any sort of now-isopacket).